src/SecureStore.BeyondTrust/BeyondTrustVaultClientFactory.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace UiPath.Orchestrator.Extensions.SecureStores.BeyondTrust
+{
+    public class BeyondTrustVaultClientFactory
+    {
+        private static BeyondTrustVaultClient client;
+
+        public static BeyondTrustVaultClient GetClient(string context)
+        {
+            if (client == null)
+            {
+                var config = JsonConvert.DeserializeObject<Dictionary<string, object>>(context);
+                client = new BeyondTrustVaultClient(config);
+            }
```

Comment:
This does not guarantee a single client is ever created. If two calls to `BeyondTrustSingleSystemSecureStore.ValidateContextAsync` are made at the same time, you can reach the null check on two threads at the same time and they both proceed to create a client and then return it.